src/gsy_e/models/strategy/pv.py

```diff
@@ -221,17 +222,32 @@ def event_tick(self):
         self._settlement_market_strategy.event_tick(self)
         self._future_market_strategy.event_tick(self)
 
-    def set_produced_energy_forecast_kWh_future_markets(self, reconfigure=True):
+    def set_produced_energy_forecast_kWh_in_state(self, reconfigure=True):
+        self.set_produced_energy_forecast_kWh_spot_market(reconfigure)
+        self.set_produced_energy_forecast_kWh_future_market(reconfigure)
+
+    def set_produced_energy_forecast_kWh_spot_market(self, reconfigure=True):
         # This forecast is based on the real PV system data provided by enphase
         # They can be found in the tools folder
         # A fit of a gaussian function to those data results in a formula Energy(time)
         market = self.area.spot_market
         slot_time = market.time_slot
-        difference_to_midnight_in_minutes = \
-            slot_time.diff(self.area.now.start_of("day")).in_minutes() % (60 * 24)
+        difference_to_midnight_in_minutes = slot_time.diff(
+            self.area.now.start_of("day")).in_minutes() % (60 * 24)
         available_energy_kWh = self.gaussian_energy_forecast_kWh(
             difference_to_midnight_in_minutes) * self.panel_count
-        self.state.set_available_energy(available_energy_kWh, slot_time, reconfigure)
+        self.state.set_available_energy(available_energy_kWh, slot_time, True)
+
+    def set_produced_energy_forecast_kWh_future_market(self, reconfigure=True):
+        if not GlobalConfig.FUTURE_MARKET_DURATION_HOURS:
+            return
+        for slot_time in self.area.future_market_time_slots:
+            difference_to_midnight_in_minutes = slot_time.diff(
+                pendulum.datetime(
+                    slot_time.year, slot_time.month, slot_time.day)).in_minutes() % (60 * 24)
+            available_energy_kWh = self.gaussian_energy_forecast_kWh(
+                difference_to_midnight_in_minutes) * self.panel_count
+            self.state.set_available_energy(available_energy_kWh, slot_time, reconfigure)
```

Comment:
looks like this could be refactored into a method in order to reduce code duplication, OR you could follow my second proposal in https://github.com/gridsingularity/gsy-e/pull/1389/files#r810905663